src/dl_types.h

```diff
@@ -3,6 +3,8 @@
 #ifndef DL_DL_TYPES_H_INCLUDED
 #define DL_DL_TYPES_H_INCLUDED
 
+#define DL_DEBUGGING_ENABLED
```

Comment:
Don't define this all the time, only define it when it make sense... I.e. msvc only? Don't know if natvis works with clang?
How do we best handle "shipping builds" where we don't want this enabled?
The easiest thing might be to define the enable-define to 0 if it is not defined on the commanline?

Reply:
This is not compiler dependent, but debugger dependent. I've remote debugged gcc code on Linux with the VS IDE, I've debugged clang code on Orbis etc. So it isn't as easy to just check the compiler. But I can cerainly default it to be turned off.

Reply:
Though there seems to be bugs for gcc's STL: https://developercommunity.visualstudio.com/t/Natvis-work-incompletely-during-CMake-C/10496058